rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/520_fetch_fields.yml

```diff
@@ -0,0 +1,142 @@
+        body:
+          stored_keyword: "stored_keyword_value"
+          keyword: "keyword_value"
+          stored_value: 10
+          value: 100
+          ignored_keyword: "foobar"
+          ignored_value: foobar
+
+---
+fetch stored fields:
+
+  - do:
+      search:
+        index: test
+        body:
+          stored_fields: [ stored_keyword, stored_value, keyword, value ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0.fields.stored_keyword.0: "stored_keyword_value" }
+  - match: { hits.hits.0.fields.stored_value.0: 10 }
+  - match: { hits.hits.0.fields.keyword: null }
+  - match: { hits.hits.0.fields.value: null }
+
+---
+fetch fields:
+
+  - do:
+      search:
+        index: test
+        body:
+          fields: [ stored_keyword, stored_value, keyword, value ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0.fields.stored_keyword.0: "stored_keyword_value" }
+  - match: { hits.hits.0.fields.stored_value.0: 10 }
+  - match: { hits.hits.0.fields.keyword.0: "keyword_value" }
+  - match: { hits.hits.0.fields.value.0: 100 }
+
+---
+fetch fields and stored fields:
+
+  - do:
+      search:
+        index: test
+        body:
+          fields: [ keyword, stored_value ]
+          stored_fields: [ stored_keyword, value ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0.fields.stored_keyword.0: "stored_keyword_value" }
+  - match: { hits.hits.0.fields.stored_value.0: 10 }
+  - match: { hits.hits.0.fields.keyword.0: "keyword_value" }
+  - match: { hits.hits.0.fields.value: null }
+
+---
+fetch _ignored via stored_fields:
+
+  - do:
+      search:
+        index: test
+        body:
+          stored_fields: [ _ignored ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0.fields._ignored: null }
+  - match: { hits.hits.0._ignored.0: "ignored_keyword" }
+  - match: { hits.hits.0._ignored.1: "ignored_value" }
+
+---
+fetch _ignored via fields:
+
+  - do:
+      search:
+        index: test
+        body:
+          fields: [ _ignored ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0.fields._ignored.0: "ignored_keyword" }
+  - match: { hits.hits.0.fields._ignored.1: "ignored_value" }
+  - match: { hits.hits.0._ignored.0: "ignored_keyword" }
+  - match: { hits.hits.0._ignored.1: "ignored_value" }
+
+---
+fetch _seq_no via stored_fields:
+
+  - do:
+      search:
+        index: test
+        body:
+          stored_fields: [ _seq_no ]
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0.fields._seq_no: null }
+  - match: { hits.hits.0._seq_no: null }
+
+---
+fetch _seq_no via fields:
+
+  - do:
+      catch: "request"
+      search:
+        index: test
+        body:
+          fields: [ _seq_no ]
+
+  # This should be `unauthorized` (401) or `forbidden` (403) or at least `bad request` (400)
+  # while instead it is mapped to an `internal_server_error (500)`
+  - match: { status: 500 }
+  - match: { error.root_cause.0.type: unsupported_operation_exception }
```

Comment:
Thanks for adding all these tests, they help a lot.

Can you add one as requested [here](https://github.com/elastic/elasticsearch/pull/106325#discussion_r1553136287), that requests `fields` as well as `stored_fields: _none_`? Could you check the current behaviour on it and make sure that stays the same after your change?

Reply:
the suggested additional test is redundant, like we've seen in the other comment below. We are good without.